src/server/req_quejob.c

```diff
@@ -879,6 +880,15 @@ req_quejob(struct batch_request *preq)
 						free(pj->ji_wattr[(int)JOB_ATR_outpath].at_val.at_str);
 						pj->ji_wattr[(int)JOB_ATR_outpath].at_val.at_str = result;
 					}
+				} else if (pj->ji_wattr[(int)JOB_ATR_outpath].at_val.at_str[l -1] == '/') {
+					namebuf[0] = 0;
+					spool_filename(pj, namebuf, JOB_STDOUT_SUFFIX);
+					if ((job_name = malloc(l+strlen(namebuf)+1))) {
```

Comment:
I don't think you need this whole block of code. You can use attribute's setter (set_str) function with INCR operator which will do the same thing for you.

Reply:
I was not aware of set_str(). Thanks for suggesting this. It operates on two attributes, but my requirement is to append a string into an existing attribute value.

Reply:
It would be really good if we can avoid duplication of code and in order to do that I'd suggest to create a temporary attribute on stack. something like
```
attribute temp;
temp.at_flags |= ATR_VFLAG_SET;
temp.at_val.st_str = namebuf;
job_attr_def[JOB_ATR_outpath].at_set(&pj->ji_wattr[(int)JOB_ATR_outpath], &temp, INCR) 
```

Reply:
make sense. I've changed these.